src/Configuration/CFGCompiler/CFGCompiler.cpp

```diff
@@ -30,21 +35,31 @@ static bool programmer_flow(CFGCompiler* cfgcompiler, int argc,
   // Each command has different checking rule
   // Set this argument in CFGCompiler
   // return if there is an error
-  bool status = true;
-  cfgcompiler->m_cmdarg.command = "programmer";
+  bool status{true};
+  std::vector<std::string> errors;
+  cfgcompiler->m_cmdarg.command = "program_device";
+  auto compiler = cfgcompiler->GetCompiler();
+  const std::type_info& compilerType = typeid(*compiler);
```

Comment:
why it is necessary to use it? why now compiler name?

Reply:
I need to find a way to know what type of compiler it is, whether it is Compiler base class or openFPGACompiler or RsCompiler. 

Initially, I used cmdline input --compiler as the name of the compiler, but it cannot work probably because the reason below:
When the compiler object is instantiated in Raptor main, the default empty string "" refer to RsCompiler. But for FOEDAG, the default empty strong "" for compiler flag refer to Compiler class. 

Do you have some idea?

Reply:
You can do the following: 
Add one more field into `Compiler::m_name` and method `Compiler::Name() const`

Update Compiler constructor:
`m_name = "dummy"`
Update CompilerFPGA constructor:
`m_name = "openfpga"`
Update CompilerRS constructor:
`m_name = "compilerrs"`

Reply:
Thanks for the idea

Reply:
The CompilerRs will be updated in another PR.

Reply:
Please update all constructors. E.g. not all Compiler constructors updated

Reply:
Added in this Compiler constructor too. 
```
Compiler::Compiler(TclInterpreter* interp, std::ostream* out,
                   TclInterpreterHandler* tclInterpreterHandler)
    : m_interp(interp),
      m_out(out),
      m_tclInterpreterHandler(tclInterpreterHandler) {
  if (m_tclInterpreterHandler) m_tclInterpreterHandler->setCompiler(this);
  SetConstraints(new Constraints{this});
  IPCatalog* catalog = new IPCatalog();
  m_IPGenerator = new IPGenerator(catalog, this);
  m_simulator = new Simulator(m_interp, this, m_out, m_tclInterpreterHandler);
  m_name = "dummy";
}
```